src/clipdel.c

```diff
@@ -24,17 +25,24 @@ enum delete_mode {
 struct clipdel_state {
     enum delete_mode mode;
     bool invert_match;
-    regex_t rgx;
+    bool literal_match;
+    union {
+        regex_t rgx;
+        const char *needle;
+    };
 };
 
 /**
  * Callback for cs_remove. In order for the delete to actually happen, we must
  * be running DELETE_REAL.
  */
-static enum cs_remove_action _nonnull_
-remove_if_rgx_match(uint64_t hash _unused_, const char *line, void *private) {
+static enum cs_remove_action _nonnull_ remove_if_match(uint64_t hash _unused_,
+                                                       const char *line,
+                                                       void *private) {
     struct clipdel_state *state = private;
-    int ret = regexec(&state->rgx, line, 0, NULL, 0);
+    int ret = state->literal_match
```

Comment:
patch looks good, my only thought is that the nested ternary isn't the easiest to read. maybe just do it in two if branches? then you don't need to coerce to REG_NOMATCH, too.